SwiftRuntimeLibrary/SwiftMarshal/StructMarshal.cs

```diff
@@ -26,6 +26,8 @@ public static unsafe IntPtr RetainSwiftObject (ISwiftObject obj)
 			if (obj == null)
 				return IntPtr.Zero;
 			byte* valueBuffer = stackalloc byte [3 * IntPtr.Size];
+			if (obj.SwiftObject == IntPtr.Zero)
+				throw new SwiftRuntimeException ("SwiftObject handle is IntPtr.Zero, likely because it was disposed.");
```

Comment:
minor: I'd put checks/exceptions before any work/allocations (not that it would leak in this case, nor that performance would suffer)

Reply:
👍